src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version10/GdsBlob.cs

```diff
@@ -216,7 +300,142 @@ protected override async ValueTask GetSegmentAsync(Stream stream, CancellationTo
 		}
 	}
 
-	protected override void PutSegment(byte[] buffer)
+	internal override byte[] GetSegment()
+	{
+		var requested = SegmentSize;
+
+#if !(NET48 || NETSTANDARD2_0)
+		var tmp = ArrayPool<byte>.Shared.Rent(requested * 2);
+#else
+		var tmp = new byte[requested * 2];
+#endif
+
+		try
+		{
+			_database.Xdr.Write(IscCodes.op_get_segment);
+			_database.Xdr.Write(_blobHandle);
+			_database.Xdr.Write(requested < short.MaxValue - 12 ? requested : short.MaxValue - 12);
+			_database.Xdr.Write(DataSegment);
+			_database.Xdr.Flush();
+
+			var response = (GenericResponse)_database.ReadResponse();
+
+			RblRemoveValue(IscCodes.RBL_segment);
+			if (response.ObjectHandle == 1)
+			{
+				RblAddValue(IscCodes.RBL_segment);
+			}
+			else if (response.ObjectHandle == 2)
+			{
+				RblAddValue(IscCodes.RBL_eof_pending);
+			}
+
+			var buffer = response.Data;
+
+			if (buffer.Length == 0)
+			{
+				// previous	segment	was	last, this has no data
+				return Array.Empty<byte>();
+			}
+
+			var posInInput = 0;
+			var posInOutput = 0;
+
+			while (posInInput < buffer.Length)
+			{
+				var len = (int)IscHelper.VaxInteger(buffer, posInInput, 2);
+				posInInput += 2;
+
+				Array.Copy(buffer, posInInput, tmp, posInOutput, len);
+				posInOutput += len;
+				posInInput += len;
+			}
+
+			var actualBuffer = new byte[posInOutput];
+			Array.Copy(tmp, actualBuffer, posInOutput);
+
+			return actualBuffer;
+		}
+		catch (IOException ex)
+		{
+			throw IscException.ForIOException(ex);
+		}
+		finally
+		{
+#if !(NET48 || NETSTANDARD2_0)
+			ArrayPool<byte>.Shared.Return(tmp);
+#endif
+		}
+	}
+	internal override async ValueTask<byte[]> GetSegmentAsync(CancellationToken cancellationToken = default)
+	{
+		var requested = SegmentSize;
+
+#if !(NET48 || NETSTANDARD2_0)
+		var tmp = ArrayPool<byte>.Shared.Rent(requested * 2);
```

Comment:
I would remove the `ArrayPool` from this PR. It's not used anywhere, yet. It might be bigger sweep later across the codebase.

Reply:
addressed